app/models/incomeSourceDetails/viewmodels/CeasedBusinessDetailsViewModel.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.incomeSourceDetails.viewmodels
+
+import enums.IncomeSourceJourney.IncomeSourceType
+
+import java.time.LocalDate
+
+case class CeasedBusinessDetailsViewModel(tradingName: Option[String],
```

Comment:
I think tradingName has some length limitation, so you can apply these as well on the type level (this is optional);

Reply:
apply required on the level on level of the case class definition, something like:
`
case class CeasedBusinessDetailsViewModel(tradingName: Option[String], ... )
   required(tradingName.length < 35 , "Trading name must be ... ")
`
this would guaranty that view model constructed would be always correct, you just need to handle this exception correctly, but this is really something optional